### ISSUE TITLE
Options for applying log

### ISSUE THREAD

**Issue by joshlawrence** [dotmanila/pyxbackup#18]
I'm confused about the options for applying the log to a backup:
-  If I have `apply_log = 1` in my cnf file and I run the command `pyxbackup full`, is that the same thing as running `pyxbackup --apply-log full`?
-  Does `apply_log = 1` in the cnf file do anything to `pyxbackup incr`?
-  Do I have to have `apply_log` enabled _or_ run `pyxbackup --apply-log full` at some point in my backup rotation to have a backup that I can restore?

For example:  I do _not_ have `apply_log = 1` in my cnf file.  I run a full backup on Sunday and incrementals Monday and Tuesday.  My server crashes on Wednesday.  Using `pyxbackup --restore-backup=` mentioned at the bottom of the README file, can I restore this backup?

tl;dr - I'm not really sure what apply log does in any context.

**Comment by dotmanila** [comment]
If you have apply_log = 1 on your config, you will have 2 copies of backup. The first one is raw uncompressed copy where `innobackupex --apply-log --redo-only` will be executed against. This is stored on the work_dir. The second copy will be an untouched copy, optionally compressed if you have compress = 1 in your config file.

apply_log has an effect to incremental backups. When you do full, `innobackupex --apply-log --redo-only` is applied to that full copy on the `work_dir`, when incremental is taken, it will be applied to that initial full backup and so on.

`apply_log = 1` == `pyxbackup --apply-log full|incr`.

apply_log is optional but is good if you want to validate your backups as soon as they are taken to some degree. There is command `apply-last`, basically the same effect with `apply-log`, the difference is that the former is used on separate invocation i.e. you are storing backups to a different server and want to validate there. Another command is `restore-set` that allows you to fully prepare any specific backup, all you need to do after this is copy the prepared backup to your MySQL data directory, chown properly and start MySQL.

See more here https://www.percona.com/doc/percona-xtrabackup/2.1/innobackupex/incremental_backups_innobackupex.html

**Comment by dotmanila** [comment]
@joshlawrence does this help? :)

**Comment by joshlawrence** [comment]
yes jervin, I apologize for not responding.  feel free to close the issue.
 :)

On Thu, Sep 3, 2015 at 1:22 AM, Jervin R notifications@github.com wrote:

> @joshlawrence https://github.com/joshlawrence does this help? :)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/dotmanila/pyxbackup/issues/18#issuecomment-137347304.

## 

Josh Lawrence
903-490-3700 [cell]
help@onlinelendersupport.com
